src/typing/type_inference_js.ml

```diff
@@ -2123,20 +2123,29 @@ and statement cx = Ast.Statement.(
     toplevels cx elements;
 
     Env_js.pop_env ();
-
-    let exports = Scope.(match get "exports" module_scope with
+    let for_types, exports_ = Scope.(match get "exports" module_scope with
       | Some { specific=exports;_} ->
           (* TODO: what happens when other things are also declared? *)
-          exports
+         SMap.empty(* ???? *), exports
       | None ->
-          let map = SMap.map (
-            fun { specific = export; _ } -> export
-          ) module_scope.entries in
-          Flow_js.mk_object_with_map_proto cx reason map (MixedT reason)
-    ) in
+	 let for_types, nonfor_types =
+	   SMap.partition (fun _ { for_type; _ } -> for_type) module_scope.entries
+	 in
+         let map = SMap.map (
```

Comment:
nit: can you push this `let` back one char to match the indentation level of the rest of the case logic?